src/main/java/org/openjdk/jextract/impl/NameMangler.java

```diff
@@ -196,14 +201,45 @@ public Void visitTypedef(Declaration.Typedef typedef, Declaration parent) {
            JavaFunctionalInterfaceName.with(typedef, javaName);
            functionTypeDefNames.put(typedef.type(), javaName);
         }
+
+        // handle if this typedef is of a struct/union/enum etc.
+        Utils.forEachNested(typedef, d -> {
+            d.accept(this, typedef);
+        });
         return null;
     }
 
+    private String fallbackNameFor(Declaration parent, Declaration.Scoped nested) {
```

Comment:
This method is used to compute a predictable fallback name in case a struct w/o name is found for which a Java name is needed. The general logic is as follows:

* if the parent declaration does not have a function pointer associated, just return the parent name
* otherwise, try to match the nested declaration with a declared type in the function pointer. For arguments declaration we will generate `<parent>$x<num>` whereas for declaration in return position we generate `<parent>$return`.